Lib/enum.py

```diff
@@ -946,3 +942,42 @@ def _decompose(flag, value):
         # we have the breakdown, don't need the value member itself
         members.pop(0)
     return members, not_covered
+
+def global_int_repr(self):
+    return '%s.%s' % (self.__class__.__module__, self.name)
+
+def global_flag_repr(self):
+    module = self.__class__.__module__
+    if self._name_ is not None:
+        return '%s.%s' % (module, self.name)
+    value = self._value_
+    members = []
+    negative = value < 0
+    if negative:
+        value = ~value
+    for m in self.__class__:
+        if value & m._value_:
+            value &= ~m._value_
+            members.append('re.%s' % (m._name_, ))
+    if value:
+        members.append(hex(value))
+    res = '|'.join(members)
+    if negative:
+        if len(members) > 1:
+            res = '~(%s)' % (res, )
+        else:
+            res = '~%s' % (res, )
+    return res
+
+def global_enum(cls):
+    """
+    decorator that makes the repr() of an enum member reference its module
+    instead of its class; also exports all members to the enum's module's
+    global namespace
+    """
+    if issubclass(cls, Flag):
+        cls.__repr__ = global_flag_repr
+    else:
+        cls.__repr__ = global_int_repr
+    vars(sys.modules[cls.__module__]).update(cls.__members__)
```

Comment:
Does using `vars(x)` instead of `x.__dict__` buy you anything here?

Reply:
No, it's just what I'm used to using to get a class' namespace while I'm working on it.  Inside the class body I could use `locals()`, but then I have to remember if `locals()` is well-defined for classes, and `vars()` works whether I'm in the class body or in the metaclass.

I can change it if you think `.__dict__` is more appropriate.

Reply:
Please do. `vars()` is an end user feature and mostly meant for the REPL. Here we're just updating a module's dict.